test/yul_phaser/Chromosome.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <tools/yul_phaser/Chromosome.h>
+
+#include <libyul/optimiser/BlockFlattener.h>
+#include <libyul/optimiser/CommonSubexpressionEliminator.h>
+#include <libyul/optimiser/ConditionalSimplifier.h>
+#include <libyul/optimiser/ConditionalUnsimplifier.h>
+#include <libyul/optimiser/ControlFlowSimplifier.h>
+#include <libyul/optimiser/DeadCodeEliminator.h>
+#include <libyul/optimiser/EquivalentFunctionCombiner.h>
+#include <libyul/optimiser/ExpressionInliner.h>
+#include <libyul/optimiser/ExpressionJoiner.h>
+#include <libyul/optimiser/ExpressionSimplifier.h>
+#include <libyul/optimiser/ExpressionSplitter.h>
+#include <libyul/optimiser/ForLoopConditionIntoBody.h>
+#include <libyul/optimiser/ForLoopConditionIntoBody.h>
+#include <libyul/optimiser/ForLoopConditionOutOfBody.h>
+#include <libyul/optimiser/ForLoopInitRewriter.h>
+#include <libyul/optimiser/FullInliner.h>
+#include <libyul/optimiser/FunctionGrouper.h>
+#include <libyul/optimiser/FunctionHoister.h>
+#include <libyul/optimiser/LoadResolver.h>
+#include <libyul/optimiser/LoopInvariantCodeMotion.h>
+#include <libyul/optimiser/RedundantAssignEliminator.h>
+#include <libyul/optimiser/Rematerialiser.h>
+#include <libyul/optimiser/SSAReverser.h>
+#include <libyul/optimiser/SSATransform.h>
+#include <libyul/optimiser/StructuralSimplifier.h>
+#include <libyul/optimiser/UnusedPruner.h>
+#include <libyul/optimiser/VarDeclInitializer.h>
+#include <libyul/optimiser/VarNameCleaner.h>
+
+#include <libsolutil/CommonIO.h>
+
+#include <test/Options.h>
+
+using namespace std;
+using namespace solidity::yul;
+using namespace solidity::util;
+
+namespace solidity::phaser::test
+{
+
+BOOST_AUTO_TEST_SUITE(ChromosomeTest)
+
+BOOST_AUTO_TEST_CASE(test_all_steps)
+{
+	vector<string> allSteps({
```

Comment:
Why don't you use the function of the Suite?

Reply:
Good point. I didn't want to use `Chromosome::allStepNames()` because that's a part of the code I'm testing but I somehow missed that I can just use `OptimiserSuite::allSteps()` instead of hard-coding them. Also partly because it gives me a `map` and I need a `vector` but that's not something that can't be fixed with a quick loop :)

Reply:
Done.